lib/task-graph.js

```diff
@@ -149,11 +187,14 @@ function taskGraphFactory(
             }
 
             _.forEach(_.keys(taskData.waitOn), function(waitOnTask) {
-                var newWaitOnTaskKey = idMap[waitOnTask];
-                assert.ok(newWaitOnTaskKey, 'Task to wait on does not exist: ' + waitOnTask);
-                var waitOnTaskValue = taskData.waitOn[waitOnTask];
-                delete taskData.waitOn[waitOnTask];
-                taskData.waitOn[newWaitOnTaskKey] = waitOnTaskValue;
+                if (waitOnTask === "either") {
+                    _.forEach(_.keys(taskData.waitOn[waitOnTask]), function(eitherTask){
```

Comment:
Not a problem here, you can check the `_.forOwn`, it will be more simple if use at here.